test/abstractarray.jl

```diff
@@ -398,14 +398,17 @@ function test_map(::Type{TestAbstractArray})
     )
         X = typ[1:10...]
         _typ = typeof(arg_typ(one(typ)))
-        @test map(arg_typ, X) == _typ[1:10...]
+        for mapf in [map, asyncmap, pmap]
```

Comment:
Probably not a good idea to add `pmap` tests here. `runtests.jl` removes workers once they cross a memory threshold. This was one of the reasons parallel tests are being executed outside of the initial cluster. `parallel.jl` explicitly launches `parallel_exec.jl` in another cluster.

It is safer to additional `pmap` tests to `parallel_exec.jl` and test all conditions including effects of worker removal during a `pmap`. You should also modify `pmap` tests run under `JULIA_TESTFULL=1` and test them locally as part of this PR.